modules/vectorfieldvisualizationgl/include/modules/vectorfieldvisualizationgl/processors/3d/streamparticles.h

```diff
@@ -0,0 +1,136 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#pragma once
+
+#include <modules/vectorfieldvisualizationgl/vectorfieldvisualizationglmoduledefine.h>
+#include <inviwo/core/processors/processor.h>
+#include <inviwo/core/properties/ordinalproperty.h>
+#include <inviwo/core/properties/optionproperty.h>
+#include <inviwo/core/properties/transferfunctionproperty.h>
+#include <inviwo/core/ports/volumeport.h>
+#include <inviwo/core/ports/meshport.h>
+#include <modules/opengl/shader/shader.h>
+
+#include <inviwo/core/util/clock.h>
+#include <inviwo/core/util/timer.h>
+
+#include <modules/vectorfieldvisualization/ports/seedpointsport.h>
+
+namespace inviwo {
+
+/** \docpage{org.inviwo.StreamParticles, Stream Particles}
+ * ![](org.inviwo.StreamParticles.png?classIdentifier=org.inviwo.StreamParticles)
+ * Explanation of how to use the processor.
+ *
+ * ### Inports
+ *   * __<Inport1>__ <description>.
+ *
+ * ### Outports
+ *   * __<Outport1>__ <description>.
+ *
+ * ### Properties
+ *   * __<Prop1>__ <description>.
+ *   * __<Prop2>__ <description>
+ */
+
+/**
+ * \brief VERY_BRIEFLY_DESCRIBE_THE_PROCESSOR
+ * DESCRIBE_THE_PROCESSOR_FROM_A_DEVELOPER_PERSPECTIVE
+ */
+class IVW_MODULE_VECTORFIELDVISUALIZATIONGL_API StreamParticles : public Processor {
+public:
+    StreamParticles(InviwoApplication *app);
+    virtual ~StreamParticles() = default;
+
+    virtual void process() override;
+
+    virtual const ProcessorInfo getProcessorInfo() const override;
+    static const ProcessorInfo processorInfo_;
+
+private:
+    enum class SeedingSpace { Data, World };
+    VolumeInport volume_{"volume"};
+    SeedPoints3DInport seeds_{"seeds"};
+    MeshOutport meshPort_{"particles"};
+
+    TemplateOptionProperty<SeedingSpace> seedingSpace_{
+        "seedingSpace",
+        "Seeding Space",
+        {{"data", "Data", SeedingSpace::Data}, {"world", "World", SeedingSpace::World}}};
+
+    FloatProperty stepLength_{"stepLength", "Step Length", 0.01f, 0.0f, 1.0f};
+    IntProperty internalSteps_{"internalSteps", "Internal Steps", 10, 1, 100};
+
+    FloatMinMaxProperty particleSize_{
```

Comment:
What is the size based on?

Reply:
Depends on which processor is used to render them. If it is the Sphere Renderer it will be its radius while while the mesh renderer just ignores it and render it as GL_POINTS

Reply:
The radius will be determined by? Would probably be useful to choose the parameter which the radius is derived from. Same for color.